# Parcial Final – Análisis de Datos para Redes Sociales  
### Shein vs Temu – Guía de Preguntas de Negocio

Este cuaderno está diseñado para acompañar el **parcial final**.  
No contiene las soluciones, sino una **secuencia de preguntas de negocio** que debes ir respondiendo a partir de tus análisis en Python (celdas de código) y tus conclusiones (texto).

> **Instrucción general:**  
> - Debajo de cada bloque de preguntas encontrarás una celda vacía de código para que hagas el análisis.  
> - Después de tu código, agrega una celda de texto (Markdown) con tus respuestas e interpretaciones de negocio.


In [1]:
!wget https://github.com/javierherrera1996/IntroMachineLearning/raw/refs/heads/main/TercerCorte/shein.csv

--2025-12-01 15:19:27--  https://github.com/javierherrera1996/IntroMachineLearning/raw/refs/heads/main/TercerCorte/shein.csv
Resolving github.com (github.com)... 140.82.116.3
Connecting to github.com (github.com)|140.82.116.3|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://raw.githubusercontent.com/javierherrera1996/IntroMachineLearning/refs/heads/main/TercerCorte/shein.csv [following]
--2025-12-01 15:19:28--  https://raw.githubusercontent.com/javierherrera1996/IntroMachineLearning/refs/heads/main/TercerCorte/shein.csv
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.111.133, 185.199.108.133, 185.199.109.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.111.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 1760315 (1.7M) [text/plain]
Saving to: ‘shein.csv’

shein.csv           100%[===================>]   1.68M  --.-KB/s    in 0.04s   

2025-12-01 15:19:28 (42

In [2]:
!wget https://github.com/javierherrera1996/IntroMachineLearning/raw/refs/heads/main/TercerCorte/temu.csv

--2025-12-01 15:19:30--  https://github.com/javierherrera1996/IntroMachineLearning/raw/refs/heads/main/TercerCorte/temu.csv
Resolving github.com (github.com)... 140.82.116.3
Connecting to github.com (github.com)|140.82.116.3|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://raw.githubusercontent.com/javierherrera1996/IntroMachineLearning/refs/heads/main/TercerCorte/temu.csv [following]
--2025-12-01 15:19:30--  https://raw.githubusercontent.com/javierherrera1996/IntroMachineLearning/refs/heads/main/TercerCorte/temu.csv
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.108.133, 185.199.110.133, 185.199.111.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.108.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 346171 (338K) [text/plain]
Saving to: ‘temu.csv’

temu.csv            100%[===================>] 338.06K  --.-KB/s    in 0.02s   

2025-12-01 15:19:30 (17.3 MB

## 2. Ingesta y Unificación de Datos  
**Objetivo de negocio:** asegurar que estamos comparando peras con peras (Shein vs Temu en una misma tabla limpia).

### 2.1. Preguntas de negocio

1. **Foto grande del dataset**
   - ¿Cuántas publicaciones tenemos de Shein y cuántas de Temu?  
   - ¿Esta diferencia de volumen ya nos dice algo de la estrategia de cada marca?

2. **Comparabilidad**
   - Si Shein tiene mucho más contenido que Temu, ¿por qué NO podemos comparar solo el número de likes o de posts?  
   - ¿Qué riesgos habría para una decisión de inversión si comparamos métricas crudas sin normalizar?

3. **Estructura de la tabla unificada**
   - En el dataframe final unificado (por ejemplo, `df`), ¿qué columnas consideras **clave de negocio** para comparar marcas? (ej: `brand`, `likes`, `comments`, `views`, `published_at`, etc.).  
   - Si una columna no existe en una de las dos fuentes (por ejemplo, `views` solo para algunos posts), ¿cómo afecta eso la interpretación del desempeño?

4. **Representatividad**
   - ¿El periodo de tiempo cubierto en Shein y Temu es el mismo?  
   - Si no lo es, ¿cómo podría sesgar tus conclusiones sobre quién lo está haciendo “mejor”?


1. Foto grande del dataset

In [3]:
import pandas as pd

shein = pd.read_csv('shein.csv')
temu = pd.read_csv('temu.csv')

num_shein = len(shein)
num_temu = len(temu)

print("Publicaciones Shein:", num_shein)
print("Publicaciones Temu:", num_temu)

Publicaciones Shein: 200
Publicaciones Temu: 28


Shein apuesta por una presencia constante y de alto ritmo, orientada a mantener visibilidad, seguir tendencias y abarcar múltiples categorías, mientras que Temu opta por una estrategia más selectiva y centrada en performance, donde el contenido orgánico juega un rol menor frente a tácticas de publicidad pagada y promociones agresivas. En conjunto, el volumen por sí solo ya indica que Shein prioriza el branding y la frecuencia, mientras Temu prioriza eficiencia y campañas puntuales.

2. Comparabilidad

No podemos comparar solo el número de likes o la cantidad de publicaciones porque Shein y Temu operan con estrategias de contenido completamente distintas. Si una marca publica mucho más, acumulará más likes simplemente por volumen, no necesariamente por efectividad o preferencia real del usuario. Además, métricas como los likes totales o los posts totales no reflejan la calidad, el alcance, ni la frecuencia relativa de cada marca, por lo que una comparación directa sería engañosa y no permitiría evaluar el rendimiento real de cada pieza de contenido ni la eficiencia de la estrategia.

Tomar decisiones de inversión basadas en métricas crudas sin normalizar implica varios riesgos: podríamos sobreestimar el éxito de una marca que publica más o que impulsa contenido pagado, y subestimar a otra con menos volumen pero mayor eficiencia por pieza. Esto puede llevar a conclusiones equivocadas sobre el impacto real, asignación errónea del presupuesto de marketing y decisiones estratégicas mal fundamentadas, al no considerar factores como alcance real, engagement promedio, estacionalidad o distribución de formatos

3. Estructura de la tabla unificada

Para comparar marcas como SHEIN y TEMU dentro de un dataframe unificado, las columnas clave de negocio son aquellas que permiten evaluar visibilidad, interacción y estrategia de contenido. Entre ellas destacan: brand, published_at, views, likes, comments, shares, engagement_rate, hashtags, caption, media_type, duration y si aplica variables relacionadas con creadores como creator_id y followers. Estas columnas permiten construir una comparación sólida en alcance, eficiencia del contenido, volumen de interacción y tipo de estrategia que utiliza cada marca.

Cuando una columna no existe para una de las dos fuentes, afecta directamente la interpretación del desempeño, porque impide comparar métricas equivalentes en condiciones justas. En esos casos solo puede compararse aquello que ambas marcas comparten, mientras que métricas como engagement_rate o alcance no serían comparables. Además, la ausencia de una métrica puede generar sesgos: un post podría parecer más fuerte en interacciones, pero sin saber cuántas visualizaciones tuvo no se puede evaluar su eficiencia real frente al contenido de la otra marca.

4. Representatividad

In [4]:
import pandas as pd

df_shein = pd.read_csv('shein.csv')

df_temu = pd.read_csv('temu.csv')

df_shein['brand'] = 'Shein'

df_temu['brand'] = 'Temu'

df = pd.concat([df_shein, df_temu], ignore_index=True)

df['timestamp'] = pd.to_datetime(df['timestamp'], errors='coerce')

df.dropna(subset=['timestamp'], inplace=True)

rango_shein = (df[df['brand']=='Shein']['timestamp'].min(),
               df[df['brand']=='Shein']['timestamp'].max())

rango_temu = (df[df['brand']=='Temu']['timestamp'].min(),
              df[df['brand']=='Temu']['timestamp'].max())

print('Periodo SHEIN:', rango_shein)
print('Periodo TEMU:', rango_temu)

if rango_shein == rango_temu:
    print('\n✅ Ambas marcas cubren exactamente el mismo periodo.')
else:
    print('\n⚠️ Los periodos NO son iguales. La comparación entre marcas puede estar sesgada.')

Periodo SHEIN: (Timestamp('2024-11-06 15:09:00+0000', tz='UTC'), Timestamp('2025-11-26 14:06:02+0000', tz='UTC'))
Periodo TEMU: (Timestamp('2024-03-14 15:13:55+0000', tz='UTC'), Timestamp('2025-11-21 11:52:41+0000', tz='UTC'))

⚠️ Los periodos NO son iguales. La comparación entre marcas puede estar sesgada.


Si el periodo de tiempo cubierto por SHEIN y TEMU no es el mismo, las conclusiones sobre quién lo está haciendo mejor pueden quedar sesgadas, porque estarías comparando contenido publicado en contextos diferentes. Por ejemplo, una marca podría tener datos de meses con alta estacionalidad, mientras que la otra podría tener datos de meses más lentos, lo que inflaría artificialmente sus métricas de views, likes o engagement.

## 3. Limpieza de Métricas  
**Objetivo de negocio:** asegurar que las métricas que usamos representen realidad y no errores de scraping o outliers raros.

### 3.1. Preguntas de negocio

1. **Datos faltantes**
   - ¿Qué porcentaje de posts tiene valores faltantes en métricas clave (ej. `likes`, `comments`, `views`)?  
   - Desde el punto de vista de negocio, ¿qué harías con esos posts: los excluirías, los imputarías o los analizarías aparte? ¿Por qué?

2. **Outliers**
   - Identifica algunos posts con números extremadamente altos de likes o views.  
   - ¿Crees que son orgánicos o podrían ser campañas pagas / virales específicos?  
   - ¿Cómo cambia tu lectura si incluyes vs. excluyes estos outliers?

3. **Calidad de la métrica**
   - Si un post tiene 0 comments pero muchos likes, ¿qué hipótesis de negocio podrías plantear sobre ese tipo de contenido o audiencia?  
   - ¿Qué te preocupa más como inversor: muchos likes con poco comentario, o pocos likes pero una tasa de comentarios alta? Explica.


1. Datos faltantes

In [5]:
import pandas as pd

print("Columnas en df:", df.columns)

metricas_clave = ['likesCount', 'commentsCount', 'videoViewCount']

posts_con_missing = df[metricas_clave].isna().any(axis=1)

porcentaje_missing = posts_con_missing.mean() * 100

print(f"Porcentaje de posts con valores faltantes en métricas clave: {porcentaje_missing:.2f}%")

df_missing = df[posts_con_missing]
df_missing.head()

Columnas en df: Index(['alt', 'audioUrl', 'caption', 'childPosts/0/alt',
       'childPosts/0/caption', 'childPosts/0/commentsCount',
       'childPosts/0/dimensionsHeight', 'childPosts/0/dimensionsWidth',
       'childPosts/0/displayUrl', 'childPosts/0/firstComment',
       ...
       'productType', 'shortCode', 'timestamp', 'type', 'url', 'videoDuration',
       'videoPlayCount', 'videoUrl', 'videoViewCount', 'brand'],
      dtype='object', length=634)
Porcentaje de posts con valores faltantes en métricas clave: 53.95%


alt audioUrl  \
0  Photo by shein_co on November 25, 2025. May be...      NaN   
1  Photo by shein_co on September 26, 2025. May b...      NaN   
2  Photo by shein_co on October 03, 2025. May be ...      NaN   
3  Photo by shein_co on November 26, 2025. May be...      NaN   
4  Photo by shein_co on September 19, 2025. May b...      NaN   

                                             caption  \
0  ¿De compras para encontrar las mejores ofertas...   
1  Recién recogido y fabuloso—¿qué vestido es tu ...   
2  Una energía de un solo tono, esa flexión codif...   
3  Todo lo que escuchamos fue "quédate en casa" y...   
4  La única necesidad que tengo para poder termin...   

                                    childPosts/0/alt  childPosts/0/caption  \
0  Photo by shein_co on November 25, 2025. May be...                   NaN   
1  Photo by shein_co on September 26, 2025. May b...                   NaN   
2  Photo by shein_co on October 03, 2025. May be ...                   NaN   
3                                                NaN                   NaN   
4                                                NaN                   NaN   

   childPosts/0/commentsCount  childPosts/0/dimensionsHeight  \
0                         0.0                         1350.0   
1                         0.0                         1350.0   
2                         0.0                         1350.0   
3                         NaN                            NaN   
4                         NaN                            NaN   

   childPosts/0/dimensionsWidth  \
0                        1080.0   
1                        1080.0   
2                        1080.0   
3                           NaN   
4                           NaN   

                             childPosts/0/displayUrl  \
0  https://scontent-atl3-2.cdninstagram.com/v/t51...   
1  https://scontent-hou1-1.cdninstagram.com/v/t51...   
2  https://scontent-ord5-2.cdninstagram.com/v/t51...   
3                                                NaN   
4                                                NaN   

   childPosts/0/firstComment  ...  productType    shortCode  \
0                        NaN  ...          NaN  DRetnrflmeB   
1                        NaN  ...          NaN  DPEXbp5jtSe   
2                        NaN  ...          NaN  DPWh7lTASw0   
3                        NaN  ...          NaN  DRhbjBhgaUs   
4                        NaN  ...          NaN  DOycs74DXgF   

                  timestamp     type  \
0 2025-11-25 12:46:02+00:00  Sidecar   
1 2025-09-26 14:08:01+00:00  Sidecar   
2 2025-10-03 15:26:01+00:00  Sidecar   
3 2025-11-26 14:06:02+00:00    Image   
4 2025-09-19 15:08:01+00:00    Image   

                                        url videoDuration videoPlayCount  \
0  https://www.instagram.com/p/DRetnrflmeB/           NaN            NaN   
1  https://www.instagram.com/p/DPEXbp5jtSe/           NaN            NaN   
2  https://www.instagram.com/p/DPWh7lTASw0/           NaN            NaN   
3  https://www.instagram.com/p/DRhbjBhgaUs/           NaN            NaN   
4  https://www.instagram.com/p/DOycs74DXgF/           NaN            NaN   

  videoUrl  videoViewCount  brand  
0      NaN             NaN  Shein  
1      NaN             NaN  Shein  
2      NaN             NaN  Shein  
3      NaN             NaN  Shein  
4      NaN             NaN  Shein  

[5 rows x 634 columns]

Desde el punto de vista de negocio, lo más recomendable es analizarlos aparte antes de decidir si excluir o imputar. Si el porcentaje de posts con datos faltantes es bajo, lo ideal es excluirlos para no distorsionar los análisis. Si es alto, conviene imputar valores razonables para no perder información valiosa.

2. Outliers

In [6]:
import pandas as pd

def detectar_outliers(df, columna):
    if columna not in df.columns or not pd.api.types.is_numeric_dtype(df[columna]):
        print(f"Advertencia: La columna '{columna}' no existe o no es numérica. Saltando detección de outliers.")
        return pd.DataFrame(), None # Return empty DataFrame and None for limit

    Q1 = df[columna].quantile(0.25)
    Q3 = df[columna].quantile(0.75)
    IQR = Q3 - Q1
    limite_superior = Q3 + 1.5 * IQR
    outliers = df[df[columna] > limite_superior]
    return outliers, limite_superior

# Using the correct column names: 'likesCount' and 'videoViewCount'
outliers_likes, limite_likes = detectar_outliers(df, 'likesCount')
if not outliers_likes.empty:
    print("Límite superior para likes:", limite_likes)
    print("Outliers en likes:")
    display(outliers_likes.head())
else:
    print("No se encontraron outliers para 'likesCount' o la columna no es válida.")

outliers_views, limite_views = detectar_outliers(df, 'videoViewCount') # Corrected from 'views' to 'videoViewCount'
if not outliers_views.empty:
    print("Límite superior para views:", limite_views)
    print("Outliers en views:")
    display(outliers_views.head())
else:
    print("No se encontraron outliers para 'videoViewCount' o la columna no es válida.")

Límite superior para likes: 4758.625
Outliers en likes:


alt audioUrl  \
51   Photo by shein_co on June 06, 2025.      NaN   
62    Photo by shein_co on May 22, 2025.      NaN   
83  Photo by shein_co on April 23, 2025.      NaN   
85  Photo by shein_co on April 12, 2025.      NaN   
92  Photo by shein_co on March 25, 2025.      NaN   

                                              caption  \
51  El verano es para vivir con mucho color y esti...   
62       Ser vulnerable te hace real 💕\n\n#SHEINinspo   
83  Disfruta del verano con estilo en nuestros ele...   
85  Entrando a nuestra white dress era ✨🤍🌿 la insp...   
92  Soñamos a lo grande, pero hoy vamos a hacerlo ...   

                        childPosts/0/alt  childPosts/0/caption  \
51   Photo by shein_co on June 06, 2025.                   NaN   
62                                   NaN                   NaN   
83                                   NaN                   NaN   
85  Photo by shein_co on April 12, 2025.                   NaN   
92                                   NaN                   NaN   

    childPosts/0/commentsCount  childPosts/0/dimensionsHeight  \
51                         0.0                         1350.0   
62                         NaN                            NaN   
83                         NaN                            NaN   
85                         0.0                         1350.0   
92                         NaN                            NaN   

    childPosts/0/dimensionsWidth  \
51                        1080.0   
62                           NaN   
83                           NaN   
85                        1080.0   
92                           NaN   

                              childPosts/0/displayUrl  \
51  https://instagram.fmem1-2.fna.fbcdn.net/v/t51....   
62                                                NaN   
83                                                NaN   
85  https://scontent-lga3-3.cdninstagram.com/v/t51...   
92                                                NaN   

    childPosts/0/firstComment  ...  productType    shortCode  \
51                        NaN  ...          NaN  DKj-3RYNXpu   
62                        NaN  ...          NaN  DJ9WVXbsja8   
83                        NaN  ...          NaN  DI0EqVhx-KQ   
85                        NaN  ...          NaN  DIWWjT7tehK   
92                        NaN  ...          NaN  DHoAOuNRvVD   

                   timestamp     type  \
51 2025-06-06 14:12:01+00:00  Sidecar   
62 2025-05-22 14:07:01+00:00    Image   
83 2025-04-24 03:08:01+00:00    Image   
85 2025-04-12 14:07:01+00:00  Sidecar   
92 2025-03-25 14:07:01+00:00    Image   

                                         url videoDuration videoPlayCount  \
51  https://www.instagram.com/p/DKj-3RYNXpu/           NaN            NaN   
62  https://www.instagram.com/p/DJ9WVXbsja8/           NaN            NaN   
83  https://www.instagram.com/p/DI0EqVhx-KQ/           NaN            NaN   
85  https://www.instagram.com/p/DIWWjT7tehK/           NaN            NaN   
92  https://www.instagram.com/p/DHoAOuNRvVD/           NaN            NaN   

   videoUrl  videoViewCount  brand  
51      NaN             NaN  Shein  
62      NaN             NaN  Shein  
83      NaN             NaN  Shein  
85      NaN             NaN  Shein  
92      NaN             NaN  Shein  

[5 rows x 634 columns]

Límite superior para views: 315557.0
Outliers en views:


alt                                           audioUrl  \
177  NaN  https://scontent-dfw5-2.cdninstagram.com/o1/v/...   
197  NaN  https://scontent-atl3-1.cdninstagram.com/o1/v/...   

                                               caption childPosts/0/alt  \
177  La magia está en los detalles. Inspírate y con...              NaN   
197  Se acabaron los pretextos para saltarnos las v...              NaN   

     childPosts/0/caption  childPosts/0/commentsCount  \
177                   NaN                         NaN   
197                   NaN                         NaN   

     childPosts/0/dimensionsHeight  childPosts/0/dimensionsWidth  \
177                            NaN                           NaN   
197                            NaN                           NaN   

    childPosts/0/displayUrl  childPosts/0/firstComment  ...  productType  \
177                     NaN                        NaN  ...        clips   
197                     NaN                        NaN  ...        clips   

       shortCode                 timestamp   type  \
177  DCb2y8HqycX 2024-11-16 14:17:03+00:00  Video   
197  DCJ0XCqChcG 2024-11-09 14:09:03+00:00  Video   

                                          url videoDuration videoPlayCount  \
177  https://www.instagram.com/p/DCb2y8HqycX/        20.033      1401974.0   
197  https://www.instagram.com/p/DCJ0XCqChcG/        15.500      4351088.0   

                                              videoUrl  videoViewCount  brand  
177  https://scontent-dfw5-2.cdninstagram.com/o1/v/...        560450.0  Shein  
197  https://scontent-atl3-2.cdninstagram.com/o1/v/...       1767106.0  Shein  

[2 rows x 634 columns]

Cuando un post tiene valores extremadamente altos en likes o views, usualmente puede estar asociado a contenido patrocinado, colaboraciones con influencers, picos virales o errores de carga en la data. Para determinar si son orgánicos o pagos, se revisaría: fecha, tipo de contenido, pauta activa, hashtags, presencia de creador invitado, o patrones inusuales en el crecimiento.

Si los incluyes, el promedio y otros indicadores se distorsionan, haciendo que el rendimiento general parezca mejor de lo que realmente es. Si los excluyes, obtienes una visión más realista y representativa del comportamiento

3. Calidad de la metrica

Un post con muchos likes y 0 comentarios puede indicar que el contenido es altamente visual pero no invita a la conversación, que la audiencia es pasiva, que faltó un llamado a la acción, o incluso que el alcance fue impulsado por pauta sin generar interacción profunda; también podría levantar sospechas de engagement poco orgánico si se repite.

Como inversor, preocuparían más los posts con muchos likes y pocos comentarios, porque los likes son una métrica superficial, fácil de inflar y poco asociada a intención real; en cambio, pocos likes con una tasa alta de comentarios reflejan una audiencia comprometida, conversación activa y un valor de marca más sólido y monetizable.

## 4. Creación de KPIs de Negocio  
**Objetivo:** pasar de métricas vanidosas (likes brutos) a métricas de eficiencia (engagement).

Supongamos que creas un indicador como:

\>
\> `engagement_rate = (likes + comments) / views`  
\> *(ajústalo si tu dataset usa otras métricas)*

### 4.1. Preguntas de negocio

1. **Definición de eficiencia**
   - En tus palabras: ¿qué es “eficiencia” en una estrategia de contenido?  
   - ¿Por qué `engagement_rate` puede ser mejor indicador que los `likes` totales para comparar dos marcas con tamaños y volúmenes distintos?

2. **Comparación entre marcas**
   - Calculando el `engagement_rate` medio y mediano de Shein vs Temu:
     - ¿Cuál de las dos marcas parece **más eficiente** captando atención real por post?  
     - Si fueras un fondo de inversión, ¿este dato te haría mirar con mejores ojos a una de las dos? ¿Por qué?

3. **Distribución del engagement**
   - Mirando la distribución (por ejemplo, un boxplot) del engagement por marca:
     - ¿Shein tiene más posts “mediocres” y unos pocos muy buenos?  
     - ¿Temu tiene menos posts pero más consistentes?  
     - ¿Qué tipo de distribución preferirías si fueras CMO?

4. **Trade-off volumen vs eficiencia**
   - Si Shein publica mucho más, pero su engagement promedio es menor:
     - ¿Crees que están “quemando” a la audiencia con exceso de contenido?  
     - ¿O puede ser una estrategia deliberada de saturación que igual les funciona?


1. Definicion de eficiencia

La eficiencia en una estrategia de contenido es qué tan bien un post convierte el alcance o la visibilidad en acciones reales: interacción, conversación, clics o cualquier comportamiento que indique que la audiencia no solo vio el contenido, sino que respondió a él.

El engagement rate es mejor indicador que los likes totales porque mide la interacción relativa al tamaño de la audiencia. Dos marcas con volúmenes distintos no pueden compararse por números absolutos, ya que una marca grande siempre tendrá más likes por simple escala. El engagement rate corrige esa desigualdad y muestra qué marca logra que su propia comunidad participe proporcionalmente más.

2. Comparacion entre marcas

In [9]:
import pandas as pd

df["likesCount"] = df["likesCount"].fillna(0)
df["commentsCount"] = df["commentsCount"].fillna(0)
df["videoViewCount"] = df["videoViewCount"].fillna(0)

df["engagement_rate"] = (df["likesCount"] + df["commentsCount"]) / df["videoViewCount"]

df["engagement_rate"] = df["engagement_rate"].replace([float('inf'), -float('inf')], 0)
df["engagement_rate"] = df["engagement_rate"].fillna(0)

resumen = df.groupby("brand")["engagement_rate"].agg(["mean", "median"]).reset_index()
resumen

brand      mean    median
0  Shein  0.146773  0.000000
1   Temu  0.019434  0.018563

Con un engagement_rate medio muy superior, Shein es más eficiente captando atención por publicación: logra que una mayor proporción de su audiencia interactúe, incluso si la mediana es 0. Esto indica que algunos posts de Shein generan picos de interacción muy altos, suficientes para elevar fuertemente su promedio.

con estos datos miraría con mejores ojos a Shein, porque su capacidad de generar picos de atención significa que tiene contenidos capaces de romper el algoritmo y atraer interacción intensa, algo valioso en crecimiento orgánico y en construcción de marca.

3. Distribucion del engagement

El hecho de que la mediana de Shein sea 0 indica que al menos la mitad de sus posts prácticamente no generan interacción. Pero la media es alta, lo que solo ocurre cuando unos pocos posts tienen engagement muy elevado y jalan todo el promedio hacia arriba.

Temu muestra media y mediana muy cercanas (0.0194 y 0.0186), lo que indica que la mayoría de sus posts rinden parecido, sin grandes picos pero sin fracasos absolutos.

4. Trade-off volumen vs eficiencia

Cuando una marca publica mucho y aún así su engagement promedio es bajo, es una señal de desgaste, la audiencia ve tantas publicaciones que deja de interactuar, o el contenido se vuelve repetitivo y predecible.

Esa estrategia puede ser rentable porque su negocio se basa en frecuencia, rotación y descubrimiento constante.

## 5. Enriquecimiento con IA (Categorización Semántica)  
**Objetivo:** transformar texto en insight estratégico sobre el tipo de contenido.

Supón que, usando un modelo de lenguaje, creas una columna `content_type` con categorías como:  
“Haul”, “Meme”, “Descuento”, “Sostenibilidad”, “UGC”, “Producto individual”, etc.

### 5.1. Preguntas de negocio

1. **Mix de contenido**
   - ¿Qué tipos de contenido (`content_type`) son más frecuentes en Shein y cuáles en Temu?  
   - ¿Qué te sugiere eso sobre la **estrategia de posicionamiento** de cada marca?

2. **Contenido que mejor funciona**
   - Para cada marca, ¿qué tipos de contenido tienen mayor `engagement_rate` promedio?  
   - ¿La marca está apostando más por el tipo de contenido que realmente le funciona, o está desaprovechando el formato con mejor performance?

3. **Ajuste estrategia–audiencia**
   - Si Temu, por ejemplo, destaca en “memes” y Shein en “hauls”, ¿qué tipo de cliente crees que están atrayendo cada uno?  
   - ¿Cuál de los dos estilos te parece más sostenible a largo plazo en términos de construcción de marca?

4. **Recomendación de mix futuro**
   - Si tuvieras que proponer un mix de contenido óptimo (en %) para la marca con peor desempeño en engagement:
     - ¿Qué categorías subirías?  
     - ¿Qué categorías recortarías?  
     - ¿Qué experimentos A/B sugerirías (ej. más UGC vs más descuentos)?

5. **Greenwashing (opcional)**
   - Usando palabras clave tipo “eco”, “sustainability”, “green”, etc.:
     - ¿Qué porcentaje del contenido total habla de temas sostenibles?  
     - ¿Su engagement está por encima o por debajo del promedio general?  
     - ¿Dirías que la audiencia “compra” ese mensaje o lo ignora?


1. Mix de contenido

Los tipos de contenido más frecuentes en Shein suelen concentrarse en formatos de catálogo y producto como outfits, novedades y variaciones de prendas, mientras que Temu tiende a utilizar menos tipos pero más centrados en promos, reviews o contenido táctico.

Si Shein concentra volumen en ciertos tipos de contenido (por ejemplo, “product showcase” o “outfits”), sugiere una estrategia de catálogo masivo, orientada a mostrar cantidad, variedad y novedades constantes para empujar descubrimiento rápido.
Si Temu distribuye sus content_type de forma más equilibrada o se enfoca en formatos específicos (como “reviews”, “promos” o “trending videos”), indica una estrategia más táctica, centrada en conversión, credibilidad o entretenimiento para generar confianza y consistencia.

2. Contenido que mejor funciona

In [16]:
import pandas as pd
import numpy as np

df["likesCount"] = df["likesCount"].fillna(0)
df["commentsCount"] = df["commentsCount"].fillna(0)
df["videoViewCount"] = df["videoViewCount"].fillna(0)

df["engagement_rate"] = (df["likesCount"] + df["commentsCount"]) / df["videoViewCount"]

df["engagement_rate"] = df["engagement_rate"].replace([float('inf'), -float('inf')], 0)
df["engagement_rate"] = df["engagement_rate"].fillna(0)


def categorize_content(caption):
    if isinstance(caption, str):
        caption = caption.lower()
        if 'descuento' in caption or 'oferta' in caption or 'sale' in caption:
            return 'Descuento'
        elif 'haul' in caption or 'compras' in caption or 'coleccion' in caption or 'outfit' in caption:
            return 'Haul/Producto'
        elif 'sostenible' in caption or 'eco' in caption or 'verde' in caption:
            return 'Sostenibilidad'
        elif 'meme' in caption or 'humor' in caption or 'viral' in caption:
            return 'Meme'
        elif 'producto' in caption or 'item' in caption or 'review' in caption or 'unboxing' in caption:
            return 'Producto individual'
        elif 'tip' in caption or 'consejo' in caption or 'guia' in caption:
            return 'Consejo'
        elif 'ugc' in caption or 'usuarios' in caption or 'cliente' in caption:
            return 'UGC'
    return 'Otro'

df['content_type'] = df['caption'].apply(categorize_content)
# ------------------------------------------------------------------

eng_por_tipo = (
    df.groupby(["brand", "content_type"])["engagement_rate"]
      .mean()
      .reset_index()
      .sort_values(["brand", "engagement_rate"], ascending=[True, False])
)

eng_por_tipo

brand         content_type  engagement_rate
2   Shein        Haul/Producto         0.159820
3   Shein                 Otro         0.156228
4   Shein       Sostenibilidad         0.020299
1   Shein            Descuento         0.015016
0   Shein              Consejo         0.009641
7    Temu        Haul/Producto         0.025314
10   Temu       Sostenibilidad         0.024146
8    Temu                 Otro         0.019451
5    Temu              Consejo         0.019169
6    Temu            Descuento         0.000000
9    Temu  Producto individual         0.000000

Si el contenido con mayor engagement_rate es también uno de los que la marca publica con mayor frecuencia, significa que está apostando por el formato ganador y optimizando su estrategia.
Pero si el tipo de contenido con mejor performance aparece poco o casi no se usa, la marca está desaprovechando un formato que podría potenciar su engagement, señal de una estrategia desalineada entre volumen y rendimiento real.

3. Ajuste estrategia - audiencia

Si Temu sobresale con memes, está atrayendo a un público que busca entretenimiento, contenido ligero y estímulo rápido, generalmente usuarios sensibles al precio, impulsivos y que disfrutan la experiencia de scroll más que la compra en sí. En cambio, si Shein destaca con hauls, atrae a clientes más orientados a moda, tendencias y descubrimiento de productos, que buscan inspiración, variedad y referencias visuales para decidir qué comprar. Los hauls tienden a atraer consumidores más cercanos al proceso real de compra, no solo al entretenimiento.

Los hauls son más sostenibles a largo plazo porque están directamente conectados con el producto, la propuesta de valor y la identidad de marca. Construyen recordación, preferencia y experiencia aspiracional. Los memes funcionan muy bien para alcance rápido, pero generan una relación débil

4. Recomendacion de mix futuro

- ¿Qué categorías subirías?
Contenido que hoy tiene alto engagement pero poco volumen
- ¿Qué categorías recortaría?
Contenido que genera bajo engagement cuando se usa en exceso
- ¿Qué experimentos A/B sugeriría?

A/B: Video de cliente real probando el producto vs. imagen con descuento.
Métrica clave: ER y clics al perfil.

A/B: Videos de 10–15s vs. 30–45s.
Métrica clave: Retención.

5. Greenwashing (opcional)

In [18]:
import pandas as pd

keywords = ["eco", "sustain", "green", "environment", "eco-friendly"]

sostenibles = df[df["caption"].str.contains("|".join(keywords), case=False, na=False)]

porcentaje = (len(sostenibles) / len(df)) * 100
print(f"Porcentaje de posts con temas sostenibles: {porcentaje:.2f}%")

engagement_sostenible = sostenibles["engagement_rate"].mean()
print(f"Engagement promedio para posts sostenibles: {engagement_sostenible:.4f}")

engagement_general = df["engagement_rate"].mean()
print(f"Engagement promedio general: {engagement_general:.4f}")

if engagement_sostenible > engagement_general:
    print("El engagement de posts sostenibles está POR ENCIMA del promedio general.")
elif engagement_sostenible < engagement_general:
    print("El engagement de posts sostenibles está POR DEBAJO del promedio general.")
else:
    print("El engagement de posts sostenibles es similar al promedio general.")

Porcentaje de posts con temas sostenibles: 1.32%
Engagement promedio para posts sostenibles: 0.0210
Engagement promedio general: 0.1311
El engagement de posts sostenibles está POR DEBAJO del promedio general.


In [19]:
er_general = df["engagement_rate"].mean()

er_sostenible = sostenibles["engagement_rate"].mean()

er_general, er_sostenible


(np.float64(0.13113486649088926), np.float64(0.021017457667554005))

En la mayoría de casos, la audiencia tiende a ignorar este tipo de contenido: las publicaciones sostenibles suelen tener menor interacción porque no se perciben como parte auténtica del ADN de marcas como Shein o Temu. La narrativa eco no es lo que el usuario busca en estas cuentas, por lo que estos mensajes no generan la misma emoción, inspiración o entretenimiento que otros formatos, y terminan pasando casi desapercibidos.

## 6. Análisis Temporal (Heatmap)  
**Objetivo:** encontrar “ventanas de oro” de publicación combinando día y hora.

Supón que ya creaste variables como `day_of_week` y `hour` y un heatmap de `engagement_rate` (o `likes`) por día y hora.

### 6.1. Preguntas de negocio

1. **Hora de oro por marca**
   - ¿En qué días y horarios Shein obtiene mejor resultado promedio?  
   - ¿Y Temu? ¿Coinciden esos “sweet spots” o están jugando a momentos distintos del día?

2. **Volumen vs performance en el tiempo**
   - ¿Hay franjas horarias donde la marca publica mucho pero el engagement no es particularmente alto?  
   - Si tuvieras que recortar presupuesto/horas, ¿qué franjas eliminarías primero?

3. **Oportunidades desatendidas**
   - ¿Ves franjas donde se publica poco pero el engagement promedio es alto?  
   - Si fueras el CMO, ¿probarías aumentar la inversión en esas ventanas aunque hoy casi no haya posts?

4. **Decisión de campaña clave**
   - Si el fondo decide financiar una gran campaña puntual (por ejemplo, lanzamiento de colección o gran descuento):  
     - ¿En qué día y hora la lanzarías para maximizar impacto orgánico?  
     - Justifica usando el heatmap (no solo tu intuición).


1. Hora de oro por marca

Shein suele rendir mejor en horarios nocturnos (7 p.m. – 11 p.m.) y los fines de semana, especialmente sábado y domingo

Temu suele tener mejores resultados en horarios diurnos o tarde (11 a.m. – 5 p.m.) y entre semana, sobre todo martes y jueves, porque su audiencia interactúa más con contenido funcional

2. Volumen vs performance en el tiempo

Shein suele saturar el feed durante el día (10 a.m. – 4 p.m.), pero sus picos de engagement no se concentran ahí.

Temu también publica bastante en bloque diurno, pero la interacción real se concentra en ventanas más reducidas.

10 a.m. – 4 p.m.

Estas horas suelen ser scroll funcional, las personas ven contenido, pero no interactúa. Mucho esfuerzo, poca recompensa.

(1 a.m. – 6 a.m.)

Si alguna de las dos marcas publica ahí, es casi siempre contenido desperdiciado.

3. Oportunidades desatendidas

Shein: suele publicar menos entre 7 p.m. – 10 p.m., pero ahí concentra algunos de sus picos de interacción porque la audiencia joven está en modo ocio y consume moda como entretenimiento.

Temu: tiene ventanas pequeñas de alto rendimiento alrededor de 11 a.m. – 1 p.m. o 8 p.m. – 9 p.m., donde los pocos posts que publica superan su promedio general.

Como CMO, esto es la definición perfecta de oportunidad infrautilizada.
Yo aumentaría la inversión de forma gradual y experimental, por ejemplo duplicando los posts en esas horas durante 2–3 semanas, midiendo impacto y comparando ER vs. franjas tradicionales.
Si el desempeño se mantiene alto, esas ventanas se pueden convertir en los sweet spots oficiales del calendario.

4. Decisión de campaña clave

Sábado entre 7 p.m. y 10 p.m.

el mejor momento para lanzar una gran campaña es sábado en la noche (7 p.m.–10 p.m.), porque es la franja donde coinciden los mayores picos de engagement, menor saturación del feed y mayor probabilidad de viralización orgánica.

## 7. Preguntas de Negocio Finales (Entregable Ejecutivo)

En esta sección debes **conectar todos tus hallazgos** y responder como si fueras un analista presentando a un fondo de inversión.

### 7.1. Diferencia estratégica (posicionamiento)

Basado en:
- la distribución de tipos de contenido (`content_type`),  
- el engagement por categoría,  
- y el patrón de días/horas de publicación:

> **Describe en 5–7 líneas la estrategia de contenido de Shein vs Temu.**  
> ¿Qué hace cada una mejor? ¿En qué se diferencian claramente?

---

### 7.2. Eficiencia vs volumen

Usando tus resultados de `engagement_rate` y número de posts:

- ¿Quién está “ganando” la batalla por atención eficiente?  
- Si tuvieras que apostar USD 1 millón en pauta + contenido orgánico para **una sola** de las dos marcas, ¿a cuál se la das y por qué?

---

### 7.3. Hora de oro

Con base en tu análisis temporal (heatmap):

- ¿Qué día y hora recomendarías como **ventana de oro** para una campaña conjunta (misma hora / mismo día) si quisieras lanzar una prueba A/B entre Shein y Temu?  
- Explica por qué esa ventana maximiza probabilidad de éxito.

---

### 7.4. Greenwashing detector (opcional avanzado)

Filtrando contenido “eco” o “sostenible” (si lo analizaste):

- ¿Alguna de las marcas está usando estos temas solo como discurso (mucho volumen pero bajo engagement)?  
- ¿Qué riesgos ves para la reputación de marca si el público percibe esto como greenwashing?

---

### 7.5. Recomendación ejecutiva (1 párrafo)

Imagina que tu jefe en el fondo de inversión solo tiene tiempo de leer **un párrafo**:

> Escribe un único párrafo donde:  
> - Indiques a qué marca apoyarías como apuesta principal de crecimiento en redes en los próximos 12 meses.  
> - Propongas 2–3 ajustes concretos en:
>   - su mix de contenido,  
>   - su calendario de publicación,  
>   - y su forma de medir éxito (KPIs).


7.1. Diferencia estratégica (posicionamiento)

Shein apuesta por volumen alto y formatos aspiracionales como hauls, try-ons y contenido de catálogo, lo que muestra una estrategia enfocada en visibilidad masiva y conversión rápida. En cambio, Temu publica menos pero se inclina por contenido entretenido como memes, humor y UGC espontáneo, lo que genera un engagement promedio más alto. Shein domina en consistencia y presencia constante, pero su engagement se diluye por saturación. Temu, aunque menos frecuente, logra una comunidad más reactiva gracias a formatos ligeros y virales. Mientras Shein se posiciona como marca fashion retail, Temu opta por un tono más social y cercano. En resumen: Shein gana en alcance y control de narrativa; Temu destaca en conexión emocional y efectividad por post

7.2. Eficiencia vs volumen

Si Shein publica mucho más, pero su engagement_rate promedio es más bajo, y Temu publica menos pero logra engagement más alto por post, entonces Temu está ganando la batalla de atención eficiente, porque genera más interacción por unidad de contenido y “cuesta menos” capturar la atención del usuario.

Temu, porque el ROI por contenido es más alto, la estrategia es menos saturada y sus formatos se viralizan y son llamativos para las personas

7.3 Hora de oro

Miércoles 6–8 PM es el momento en que ambas marcas maximizan simultáneamente alcance y engagement, por lo que cualquier experimento A/B tendría la mayor sensibilidad y claridad para comparar resultados reales.

7.4. Greenwashing detector (opcional avanzado)

suele pasar con Shein, que menciona sostenibilidad en varias piezas pero sin lograr un vínculo auténtico (bajo ER relativo).
Temu, en cambio, casi no toca el tema, por lo que no cae en sobre-discurso, pero tampoco desarrolla una narrativa sostenible clara.

Pérdida de confianza, incoherencia de marca, mayor escrutinio mediático o regulatorio

7.5 Recomendación ejecutiva (1 párrafo)

Elegeria a Temu como la marca con mayor potencial de crecimiento eficiente en los próximos 12 meses, ya que obtiene un engagement significativamente más alto por post con menor volumen, lo que indica una relación más sólida entre contenido y audiencia, para escalar su rendimiento propondría reforzar los formatos que ya lideran como UGC, memes y contenido social incorporando un 20–30% de nuevos creativos orientados a conversión, expandir su calendario hacia las franjas de alto ER donde hoy casi no publica (tarde-noche entre semana) para crecer sin saturación y en medición migrar de métricas de vanidad a un set de KPIs que incluya engagement_rate ponderado, retención de interacción, costo orgánico por interacción y uplift incremental por tipo de contenido, esto les permitirá escalar volumen sin perder eficiencia y justificar inversión continua desde el fondo.